extensions/amp-animation/0.1/amp-animation.js

```diff
@@ -69,36 +68,35 @@ export class AmpAnimation extends AMP.BaseElement {
   /** @override */
   buildCallback() {
     const ampdoc = this.getAmpDoc();
+    const {element} = this;
 
     // Trigger.
-    const trigger = this.element.getAttribute('trigger');
+    const trigger = element.getAttribute('trigger');
+
     if (trigger) {
-      this.triggerOnVisibility_ = userAssert(
+      userAssert(
         trigger == 'visibility',
         'Only allowed value for "trigger" is "visibility": %s',
-        this.element
+        element
       );
-    }
 
-    // TODO(dvoytenko): Remove once we support direct parent visibility.
-    if (trigger == 'visibility') {
-      userAssert(
-        this.element.parentNode == this.element.ownerDocument.body ||
-          this.element.parentNode == ampdoc.getBody(),
-        '%s is only allowed as a direct child of <body> element when trigger' +
-          ' is visibility. This restriction will be removed soon.',
-        TAG
-      );
+      const isBodyChild =
+        element.parentNode == element.ownerDocument.body ||
+        element.parentNode == ampdoc.getBody();
+
+      if (isBodyChild) {
+        this.triggerOnVisibility_ = true;
+      } else {
+        // TODO(alanorozco): The only possible parent for
+        // amp-animation[trigger=visibility] other than <body> is
+        // <amp-story-page>. Stories coordinate animations, so we bail out in
+        // this case. For other container elements that may be supported in the
+        // future, this shouldn't short-circuit.
+        return;
+      }
     }
 
-    // Parse config.
-    const scriptElement = userAssert(
-      childElementByTag(this.element, 'script'),
-      '"<script type=application/json>" must be present'
-    );
-    this.configJson_ = tryParseJson(scriptElement.textContent, (error) => {
-      throw user().createError('failed to parse animation script', error);
-    });
+    this.configJson_ = parseAmpAnimationConfig(element);
```

Comment:
Hmm. This is quite a bit chunk cut off. It looks like we'll be essentially using this as a glorified container for `<script application/json>` tag.

Reply:
Using a different tag now.